indra/llconvexdecomposition/llconvexdecompositionvhacd.cpp

```diff
@@ -0,0 +1,492 @@
+    voxel_quality_enums[E_NORMAL_QUALITY].mValue = 400000;
+    voxel_quality_enums[E_HIGH_QUALITY].mName = "High";
+    voxel_quality_enums[E_HIGH_QUALITY].mValue = 800000;
+    voxel_quality_enums[E_VERY_HIGH_QUALITY].mName = "Very High";
+    voxel_quality_enums[E_VERY_HIGH_QUALITY].mValue = 1200000;
+    voxel_quality_enums[E_ULTRA_QUALITY].mName = "Ultra";
+    voxel_quality_enums[E_ULTRA_QUALITY].mValue = 1600000;
+    voxel_quality_enums[E_MAX_QUALITY].mName = "Maximum";
+    voxel_quality_enums[E_MAX_QUALITY].mValue = 2000000;
+
+    param.mDetails.mEnumValues.mEnumsArray = voxel_quality_enums;
+    param.mDefault.mIntOrEnumValue = 400000;
+    param.mStage = 0;
+    param.mReserved = -1;
+    mDecompParams.push_back(param);
+
+    param.mName = "Num Hulls";
+    param.mDescription = nullptr;
+    param.mType = LLCDParam::LLCD_FLOAT;
+    param.mDetails.mRange.mLow.mFloat = 1.f;
+    param.mDetails.mRange.mHigh.mFloat = MAX_HULLS;
+    param.mDetails.mRange.mDelta.mFloat = 1.f;
+    param.mDefault.mFloat = 8.f;
+    param.mStage = 0;
+    param.mReserved = -1;
+    mDecompParams.push_back(param);
+
+    param.mName = "Num Vertices";
+    param.mDescription = nullptr;
+    param.mType = LLCDParam::LLCD_FLOAT;
+    param.mDetails.mRange.mLow.mFloat = 3.f;
+    param.mDetails.mRange.mHigh.mFloat = MAX_VERTICES_PER_HULL;
+    param.mDetails.mRange.mDelta.mFloat = 1.f;
+    param.mDefault.mFloat = 32.f;
+    param.mStage = 0;
+    param.mReserved = -1;
+    mDecompParams.push_back(param);
+
+    param.mName = "Error Tolerance";
+    param.mDescription = nullptr;
+    param.mType = LLCDParam::LLCD_FLOAT;
+    param.mDetails.mRange.mLow.mFloat = 0.0001f;
+    param.mDetails.mRange.mHigh.mFloat = 99.f;
+    param.mDetails.mRange.mDelta.mFloat = 0.001f;
+    param.mDefault.mFloat = 1.f;
+    param.mStage = 0;
+    param.mReserved = -1;
+    mDecompParams.push_back(param);
+
+    for (const LLCDParam& param : mDecompParams)
+    {
+        const char* const name = param.mName;
+
+        switch (param.mType)
+        {
+        case LLCDParam::LLCD_FLOAT:
+        {
+            setParam(name, param.mDefault.mFloat);
+            break;
+        }
+        case LLCDParam::LLCD_ENUM:
+        case LLCDParam::LLCD_INTEGER:
+        {
+            setParam(name, param.mDefault.mIntOrEnumValue);
+            break;
+        }
+        case LLCDParam::LLCD_BOOLEAN:
+        {
+            setParam(name, (param.mDefault.mBool != 0));
+            break;
+        }
+        case LLCDParam::LLCD_INVALID:
+        default:
+        {
+            break;
+        }
+        }
+    }
+}
+
+LLConvexDecompositionVHACD::~LLConvexDecompositionVHACD()
+{
+    mBoundDecomp = nullptr;
+    mDecompData.clear();
+
+    mVHACD->Release();
+}
+
+void LLConvexDecompositionVHACD::genDecomposition(int& decomp)
+{
+    int new_decomp_id = static_cast<int>(mDecompData.size()) + 1;
+    mDecompData[new_decomp_id] = LLDecompData();
+    decomp = new_decomp_id;
```

Comment:
Why not
```suggestion
    decomp = static_cast<int>(mDecompData.size()) + 1;
    mDecompData[decomp] = LLDecompData();
```
Does the order matter here?